### PR TITLE
Do not open dispatchers in follower role

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

In follower role, StreamProcessor is running only in replay mode. When the writer is created, dispatcher is also opened which allocates direct buffer. This is unnecessary as the writer is never used. The allocated buffer consumes memory and can create memory pressure on the system. To fix this, we create the writer only after the replay is completed. 

## Related issues

related #8509 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [x] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by deepthidevaki** [comment]
Result of the benchmark
https://github.com/camunda/zeebe/issues/8509#issuecomment-1124953094

**Comment by Zelldon** [comment]
@deepthidevaki I think it makes sense to run this test on the long running benchmark, here we see more the OOM.

**Comment by deepthidevaki** [comment]
> I think it makes sense to run this test on the long running benchmark, here we see more the OOM.

Yes. I will start a long running benchmark after the PR is merged.

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java [hunk 1]
```diff
@@ -37,6 +37,31 @@
...
+                                      |                                            |                      |
+                                      |                                            |                      |
+                                      v                                            |                      |
+                                +-------------+                                    |                      |
+                                |   Actor     |                                    |                      |
+                                |   Failed    |  <---------------------------------------------------------
+                                |             |
+                                +-------------+
+
+https://textik.com/#f8692d3c3e76c699
```
Comment (Zelldon) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java:63]: :+1: Cool :D

Comment (Zelldon) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java:63]: Maybe we add as well ActorStarted after reader creation, to make it clear it is before started :shrug:

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java [hunk 2]
```diff
@@ -304,19 +341,10 @@ private ZeebeDbState recoverState() {
...
-    logStream.registerRecordAvailableListener(this);
-
-    // start reading
-    lifecycleAwareListeners.forEach(l -> l.onRecovered(processingContext));
-    processingStateMachine.startProcessing(lastProcessingPositions);
-    if (!shouldProcess) {
-      setStateToPausedAndNotifyListeners();
-    }
+    actor.runOnCompletion(
+        logStream.newLogStreamBatchWriter(),
```
Comment (Zelldon) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java:345]: :wrench: We can also use         logStream.newLogStreamBatchWriter().onComplete

**Review by ChrisKujawa — APPROVED** [review]
Thanks @deepthidevaki :rocket::+1:

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java (hunk 1 above)
Comment (deepthidevaki) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java:63]: :+1:

**Comment by deepthidevaki** [comment]
bors merge

**Comment by oleschoenburg** [comment]
Maybe worth to backport it to at least 8.0?

**Pull request merged**